### ISSUE TITLE
Octoprint requires Raspbian Jessie based distro to work with Raspberry Pi 3

### ISSUE THREAD

**Issue by alex-gibson** [guysoft/OctoPi#207]
I have been testing ahead of the launch today, and noticed that the standard OctoPi distro gives rainbow screen of death on the Pi 3, and I could not follow the instructions (due to errors) to install for Jessie, using latest NOOBS distro as at last night.

For hassle free OctoPrint on the Pi 3 I recommend starting with the latest Jessie nightly build, which worked well enough that I demo'd it at the Pi 3 launch party today :)

Happy to contribute testing to get a stable Jessie/Pi3 build!!!

**Comment by taxilian** [comment]
I just ran into this as well; I appreciate knowing that I'm not the only one, and I'll proceed following those instructions =]

**Comment by nophead** [comment]
I have OctoPrint running on a RPi3 by installing latest NOOBS and then following the OctoPrint install instructions for Linux. Tedious but they work and they also work on Ubuntu on an Orange Pi.

**Comment by gammy** [comment]
I installed Raspbian Jessie Lite (2016-02-26), but did so manually rather than via NOOBS. I then followed the OctoPrint installation instructions, except that I checked out v1.2.9(the latest stable version as of writing) rather than using the master. Tedious indeed @nophead! It all works though, including optional dependencies such as mjpg-streamer and cura-engine.

**Comment by nophead** [comment]
What it needs is an OctoPrint install script that runs on any Linux and does what the Linux instructions specify. Then you don't really need OctoPi.

**Comment by guysoft** [comment]
Hey guys, have you tried the nightly build?
http://docstech.net/OctoPiMirror/nightly/
eg.
https://storage.googleapis.com/octoprint/2016-03-02_2016-02-26-octopi-jessie-lite-0.13.0.zip

**Comment by pmbdk** [comment]
The latest nightly build works fine with RPi3 (although I havn't actually tried printing with it yet)

**Comment by A-Nesh** [comment]
I have also installed a nightly build (2016-03-02_2016-02-26-octopi-jessie-lite-0.13.0.zip) in RPi 3. The RPi freezes during a print.

**Comment by seefood** [comment]
Wow, people got their RP3 quickly :-)
Now @guysoft only needs to take care of a Pine64 version and.... Hey, isn't it time to just create a PPA? :)

**Comment by foosel** [comment]
> The Pi freezes half way through a print.

I've run into a similar issue with the Pi3 I just got, mine just sat idly though and suddenly stopped responding altogether, several times. On first look it seems to be power related, switched to a different PSU, haven't been able to reproduce the issue since then (still observing it though). 

Since the Pi3 [does consume more power than the Pi2](https://www.element14.com/community/community/raspberry-pi/blog/2016/02/29/the-most-comprehensive-raspberry-pi-comparison-benchmark-ever), it might be that one simply can't get away with the same power supplies anymore as was still possible with Pi1/Pi2. Just an idea though. I certainly never had a Pi2 just freeze on me with that PSU while doing basically nothing.

**Comment by A-Nesh** [comment]
I don't believe that it's a power issue in my case as mine PPi is connected to a ATX PSU which can deliver 25A on a 5V rail.

**Comment by foosel** [comment]
After it just happened again, I have to agree, doesn't seem to be power related.

I also have to correct myself, it apparently doesn't freeze completely after all, it just looses wifi connectivity. Funny thing is, it believes it still does have connectivity, has an IP and everything, but cannot be pinged or SSH'd into. Until I ping the router from the Pi3, then suddenly connectivity is back:

```
$ ping -t 192.168.1.204

Pinging 192.168.1.204 with 32 bytes of data:
Reply from 192.168.1.3: Destination host unreachable.
Reply from 192.168.1.3: Destination host unreachable.
Reply from 192.168.1.3: Destination host unreachable.
Reply from 192.168.1.3: Destination host unreachable.
Reply from 192.168.1.204: bytes=32 time=2030ms TTL=64
```

Happens very reproducibly after some idle time. Maybe some power safe feature of the wifi chipset that's misbehaving (like it also happens with some dongles out there)?

@A-Nesh did it really freeze for you during printing (print stopped) or could it be that just the network connection broke down (print didn't stop, but anything relying on network didn't work anymore, so OctoPrint, SSH, ...).

**Comment by A-Nesh** [comment]
@foosel my printer just stopped printing, after some printing (an hour or so). The extruder and the heated bed was still on. It happened two times today. I don't know if the whole Pi freeze or not but it was not accessible either by HTTP or SSH. I'll check it next time it misbehave.

**Comment by foosel** [comment]
If it even stopped printing, that sounds like a different issue then. We are running the same image, so if it's an image issue, I should hopefully see it too during my tests.

From what you describe it indeed sounds like the whole Pi crashes completely. Do you have a chance to attach a monitor to it? That might show what's happening when it goes down. Also of interest would be your `/var/log/syslog` and you `dmesg`, maybe something got logged there when it went down.

**Comment by A-Nesh** [comment]
i have now encountered same issue as you @foosel. It did not stop printing but the network died. I'll connect a monitor next time it happens.

Here are my syslog and dmesg:
[dmesg.txt](https://github.com/guysoft/OctoPi/files/160374/dmesg.txt)
[syslog.txt](https://github.com/guysoft/OctoPi/files/160376/syslog.txt)

**Comment by A-Nesh** [comment]
It died again now, after 77%

PING 192.178.1.88 (192.178.1.88): 56 data bytes
Request timeout for icmp_seq 0
Request timeout for icmp_seq 1
Request timeout for icmp_seq 2
Request timeout for icmp_seq 3
Request timeout for icmp_seq 4

**Comment by A-Nesh** [comment]
I have connected a monitor to the RPi, there is no signal = the RPi is dead

**Comment by foosel** [comment]
[I asked in the G+ community](https://plus.google.com/u/0/+GinaH%C3%A4u%C3%9Fge/posts/g8jUYd8MSWY) if people were seeing this too with OctoPi and the RPi3. So far no definitive answer, but a good question:

> If they're running an ATX PSU have they got any dummy load resistors to flatten out the power? I too thought it'd be no problem but noticed under load my ATX was dropping to 4 volts. Added a dummy load and it stabilized the output. 

and some good advice I have to agree with:

> Check your power supply.  Try it without the printer turned on and see if stays up.  Try with a separate 5V power supply and running a print.  More troubleshooting is needed.
>
> Try a different memory card?﻿

Regarding the wifi going down, that's definitely caused by power management. I disabled it via `/boot/octopi-network` by adding `wireless-power off` to my `wlan0` config, right after the SSID/PSK combo ([basically as described here](http://www.modmypi.com/blog/disable-wifi-power-management)) and since doing that I couldn't reproduce the "un-ping-able" issue any more. Removed the line again and bam, issue returned. So something appears to be broken with the power management of the Pi3's own wifi.

I've seen older RPis freezing (exact same symptoms you are observing) due to issues with the wired interface, power supply issues (especially when under load), USB devices drawing too much power (which is why I prefer to have a powered hub between my Pi's and their peripherals), etc etc.

**Comment by A-Nesh** [comment]
ok, thanx. Ill try a separate power adapter and report later

**Comment by nophead** [comment]
If you are using an ATX PSU it is much better to use the 5V standby power
rather than the main 5V. It is actually a separate power supply so isn't
affected by the loads on the other outputs and you can also then use the
RPi to control the other half of the PSU.

**Comment by A-Nesh** [comment]
Have tried both 5v standby and ordinary 5v rails with no luck. I also believe that it is a power issue. I have  connected usb power monitor and can see that there almost no current consumption when the printer is on. Which means that the power must come from the printer to RPi via USB. I use geeetech gt2650 board. It was never issue with OctoPi on mi Raspberry Pi 1.

**Comment by nophead** [comment]
Power should only flow from host to device on a USB cable. If device power is powering the host it is faulty. It wouldn't surprise me from Geeetech.

**Comment by A-Nesh** [comment]
I have left the Pi over night connected to a 3A USB power supply and it was dead this morning. I have connected a monitor and as it was not responding i have restarted the Pi. A couple minutes later, i have found the issue. I am new to linux but it looks to me like file system error. 
![2016-03-07 08 34 12](https://cloud.githubusercontent.com/assets/17685372/13563252/8b7a0980-e440-11e5-85e6-fbb38c8dc4b4.jpg)

**Comment by foosel** [comment]
That looks like a kernel panic. Download /var/log/syslog (eg via Womack), upload it somewhere (Dropbox or similar), share the link here. 

The full log output of the kernel panic might tell us what's up, if it's a hardware issue, a flash gone wrong etc.

Just for the record, my pi3 has been up now since my last post yesterday without issues, using the same image and the above mentioned change to disable the Wi-Fi power management.

**Comment by taxilian** [comment]
Just as a datapoint, I installed using the raspbian image and then followed the octoprint install instructions and have been printing back to back jobs since Friday night with no issues.

Have not used Wi-Fi our changed any other settings.

**Comment by A-Nesh** [comment]
@foosel here is my syslog
[syslog.txt](https://github.com/guysoft/OctoPi/files/160947/syslog.txt)

**Comment by A-Nesh** [comment]
@taxilian actually it worked fine for me to when i installed latest Jessie and manually installed OctoPi. I don't know why did i install a nightly build instead when everything worked fine.

**Comment by foosel** [comment]
@A-Nesh because giving us early feedback on how the nightlies perform is the only way to root out issues *before* release, and you knew that and wanted to help? ;)

**Comment by A-Nesh** [comment]
@foosel yes, it's was pretty much my plan :)

**Comment by foosel** [comment]
@A-Nesh I took a look into your syslog. Seems like the issue you made a picture of above didn't make it into the log :/ But there are a LOT of errors with the wifi interface, including some traces while trying to get an IP via dhcp. Have you tried disabling WiFi power management as described above? It's just a complete shot in the dark... if that doesn't help, I'd suggest reflashing/another SD card, to rule out issues with the flash and the card.

Can't help myself, I don't have the feeling that their drivers are all "there" yet, the raspberry forums are also quite full with "doesn't work" posts in the wifi thread at the moment.

Might also be interesting to see if they maybe did things differently between the lite and the full raspbian image. OctoPi uses the lite image.

**Comment by A-Nesh** [comment]
@foosel Yes, i also notice a DHCP attempt today. It still connects to the manual IP (.88). My DHCP range is .100 an up. I also took a look in the syslog and couldn't find the error message. Maybe it died died before it could write the log. 

I have used the full Jessie image earlier. I will try to to flash a new nightly image to a new SD card, as soon i find it.  :) just to eliminate sd card issue

I didn't have a chance yet to disable wifi power management, i'll do it first now.

**Comment by A-Nesh** [comment]
I have tested fresh nightly image with wireless-power off, on a new SD card.... same issue. Here is my syslog.
[syslog.txt](https://github.com/guysoft/OctoPi/files/161264/syslog.txt)

I am now installing full Jessie image...

**Comment by A-Nesh** [comment]
Nope, same freeze during print issue. This time 54%. I give up, back to my RPi1

**Comment by foosel** [comment]
Oh, but that syslog was way more interesting:

```
Mar  7 11:56:44 octopi kernel: [ 1483.101088] Unable to handle kernel NULL pointer dereference at virtual address 00000000
Mar  7 11:56:44 octopi kernel: [ 1483.117168] pgd = 80004000
Mar  7 11:56:45 octopi kernel: [ 1483.123041] [00000000] *pgd=00000000
Mar  7 11:56:45 octopi kernel: [ 1483.130335] Internal error: Oops: 17 [#1] SMP ARM
Mar  7 11:56:45 octopi kernel: [ 1483.139454] Modules linked in: bnep hci_uart btbcm bluetooth brcmfmac brcmutil cfg80211 ftdi_sio rfkill joydev evdev usbserial snd_bcm2835 snd_pcm bcm2835_gpiomem snd_timer bcm2835_rng snd uio_pdrv_genirq uio ipv6
Mar  7 11:56:45 octopi kernel: [ 1483.175090] CPU: 1 PID: 42 Comm: kworker/1:1 Not tainted 4.1.18-v7+ #846
Mar  7 11:56:45 octopi kernel: [ 1483.187596] Hardware name: BCM2709
Mar  7 11:56:45 octopi kernel: [ 1483.194773] Workqueue: events brcmf_fweh_event_worker [brcmfmac]
Mar  7 11:56:45 octopi kernel: [ 1483.206209] task: b5b68000 ti: b5b2a000 task.ti: b5b2a000
Mar  7 11:56:45 octopi kernel: [ 1483.216669] PC is at cfg80211_get_bss_channel+0xf8/0x190 [cfg80211]
Mar  7 11:56:45 octopi kernel: [ 1483.228505] LR is at 0xfffffff6
Mar  7 11:56:45 octopi kernel: [ 1483.235206] pc : [<7f12eb48>]    lr : [<fffffff6>]    psr: 60000013
Mar  7 11:56:45 octopi kernel: [ 1483.235206] sp : b5b2bcc8  ip : 00000003  fp : b5b2bcec
Mar  7 11:56:45 octopi kernel: [ 1483.257060] r10: 00000064  r9 : 00000000  r8 : 00000000
Mar  7 11:56:45 octopi kernel: [ 1483.267106] r7 : 000000c5  r6 : 000000b4  r5 : b2d60160  r4 : b3f90095
Mar  7 11:56:45 octopi kernel: [ 1483.279271] r3 : 00000000  r2 : 000000c5  r1 : b3f90084  r0 : 0000000d
Mar  7 11:56:45 octopi kernel: [ 1483.291417] Flags: nZCv  IRQs on  FIQs on  Mode SVC_32  ISA ARM  Segment kernel
Mar  7 11:56:45 octopi kernel: [ 1483.304842] Control: 10c5383d  Table: 32da806a  DAC: 00000015
Mar  7 11:56:45 octopi kernel: [ 1483.315699] Process kworker/1:1 (pid: 42, stack limit = 0xb5b2a210)
Mar  7 11:56:45 octopi kernel: [ 1483.327372] Stack: (0xb5b2bcc8 to 0xb5b2c000)
Mar  7 11:56:45 octopi kernel: [ 1483.335969] bcc0:                   00000000 b2d60160 000000c5 000000c5 00000000 00000000
Mar  7 11:56:45 octopi kernel: [ 1483.352402] bce0: b5b2bd74 b5b2bcf0 7f12fa18 7f12ea5c 80819cdc 00000c11 00000000 00000000
Mar  7 11:56:45 octopi kernel: [ 1483.368889] bd00: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
Mar  7 11:56:45 octopi kernel: [ 1483.385386] bd20: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
Mar  7 11:56:45 octopi kernel: [ 1483.401948] bd40: 00000000 00000000 b5247c40 b2d60160 000000d0 000000c5 00000064 00000c11
Mar  7 11:56:45 octopi kernel: [ 1483.418617] bd60: 00000000 b3f90004 b5b2bde4 b5b2bd78 7f251700 7f12f9b4 b3f9000c b2d60160
Mar  7 11:56:45 octopi kernel: [ 1483.435392] bd80: 00000000 00000000 00000c11 00000064 b3f90084 000000c5 ffffeae8 000000d0
Mar  7 11:56:45 octopi kernel: [ 1483.452310] bda0: 8056be54 00000c11 00000800 7f2598a4 b5b2bde4 b5b2bdc0 7f2598a4 b3f90004
Mar  7 11:56:45 octopi kernel: [ 1483.469432] bdc0: b3f90000 b5247c40 00000000 b2d60160 b2e3b248 b2e3b000 b5b2be0c b5b2bde8
Mar  7 11:56:45 octopi kernel: [ 1483.486644] bde0: 7f252204 7f25163c b5b2be88 b2d60340 b5b2bea0 b2d60340 b5247800 b5247c40
Mar  7 11:56:45 octopi kernel: [ 1483.503926] be00: b5b2be5c b5b2be10 7f252474 7f25219c 80056f60 80056a28 ffffffff 00ffffff
Mar  7 11:56:45 octopi kernel: [ 1483.521267] be20: 00000001 b6fc0ac0 0001ce82 00000001 b6fc0ad0 00000013 b2c36098 b2c360a8
Mar  7 11:56:45 octopi kernel: [ 1483.532236] be40: 00200200 00100100 b2c360ac b2c34000 b5b2be74 b5b2be60 7f259de0 7f2523c8
Mar  7 11:56:45 octopi kernel: [ 1483.542567] be60: b2de3cc3 b2de3c80 b5b2bee4 b5b2be78 7f259ea4 7f259da4 b529a8c0 b2e8e780
Mar  7 11:56:45 octopi kernel: [ 1483.552898] be80: b6fc0580 00000002 00000002 00000013 00000000 00000001 00000000 00000000
Mar  7 11:56:45 octopi kernel: [ 1483.563230] bea0: e5ef7b10 6c77669d 00000030 00000000 00000000 00000000 80569d94 b2c36098
Mar  7 11:56:45 octopi kernel: [ 1483.573562] bec0: b5b15a00 b6fc0094 b6fc0080 b6fc3f00 00000000 00000000 b5b2bf24 b5b2bee8
Mar  7 11:56:45 octopi kernel: [ 1483.583906] bee0: 8003dbf8 7f259e48 b6fc0080 b6fc0094 b5b2a000 00000008 b5b15a00 b6fc0080
Mar  7 11:56:45 octopi kernel: [ 1483.594236] bf00: b5b15a18 b6fc0094 b5b2a000 00000008 b5b15a00 b6fc0080 b5b2bf5c b5b2bf28
Mar  7 11:56:45 octopi kernel: [ 1483.604569] bf20: 8003dec4 8003dab0 8003de70 80818900 00000000 00000000 b5978080 b5b15a00
Mar  7 11:56:45 octopi kernel: [ 1483.614898] bf40: 8003de70 00000000 00000000 00000000 b5b2bfac b5b2bf60 8004364c 8003de7c
Mar  7 11:56:45 octopi kernel: [ 1483.625229] bf60: b5b2bf94 00000000 8004a260 b5b15a00 00000000 00000000 b5b2bf78 b5b2bf78
Mar  7 11:56:45 octopi kernel: [ 1483.635560] bf80: 00000000 00000000 b5b2bf88 b5b2bf88 b5978080 80043560 00000000 00000000
Mar  7 11:56:45 octopi kernel: [ 1483.645893] bfa0: 00000000 b5b2bfb0 8000faf8 8004356c 00000000 00000000 00000000 00000000
Mar  7 11:56:45 octopi kernel: [ 1483.656229] bfc0: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
Mar  7 11:56:45 octopi kernel: [ 1483.666561] bfe0: 00000000 00000000 00000000 00000000 00000013 00000000 55555575 55555755
Mar  7 11:56:45 octopi kernel: [ 1483.676941] [<7f12eb48>] (cfg80211_get_bss_channel [cfg80211]) from [<7f12fa18>] (cfg80211_inform_bss_width+0x70/0x234 [cfg80211])
Mar  7 11:56:45 octopi kernel: [ 1483.690965] [<7f12fa18>] (cfg80211_inform_bss_width [cfg80211]) from [<7f251700>] (brcmf_inform_single_bss+0xd0/0x140 [brcmfmac])
Mar  7 11:56:45 octopi kernel: [ 1483.704867] [<7f251700>] (brcmf_inform_single_bss [brcmfmac]) from [<7f252204>] (brcmf_update_bss_info.part.8+0x74/0x130 [brcmfmac])
Mar  7 11:56:45 octopi kernel: [ 1483.719036] [<7f252204>] (brcmf_update_bss_info.part.8 [brcmfmac]) from [<7f252474>] (brcmf_notify_roaming_status+0xb8/0x1a0 [brcmfmac])
Mar  7 11:56:45 octopi kernel: [ 1483.733559] [<7f252474>] (brcmf_notify_roaming_status [brcmfmac]) from [<7f259de0>] (brcmf_fweh_call_event_handler+0x48/0xa4 [brcmfmac])
Mar  7 11:56:45 octopi kernel: [ 1483.748080] [<7f259de0>] (brcmf_fweh_call_event_handler [brcmfmac]) from [<7f259ea4>] (brcmf_fweh_event_worker+0x68/0x2d0 [brcmfmac])
Mar  7 11:56:45 octopi kernel: [ 1483.762320] [<7f259ea4>] (brcmf_fweh_event_worker [brcmfmac]) from [<8003dbf8>] (process_one_work+0x154/0x3cc)
Mar  7 11:56:45 octopi kernel: [ 1483.774520] [<8003dbf8>] (process_one_work) from [<8003dec4>] (worker_thread+0x54/0x500)
Mar  7 11:56:45 octopi kernel: [ 1483.784763] [<8003dec4>] (worker_thread) from [<8004364c>] (kthread+0xec/0x104)
Mar  7 11:56:45 octopi kernel: [ 1483.793213] [<8004364c>] (kthread) from [<8000faf8>] (ret_from_fork+0x14/0x3c)
Mar  7 11:56:45 octopi kernel: [ 1483.801556] Code: 0afffff1 e3520015 9affffef e5d10002 (e5931000) 
Mar  7 11:56:46 octopi kernel: [ 1483.810443] ---[ end trace b08645ac099d9c8f ]---
Mar  7 11:56:46 octopi kernel: [ 1483.810537] Unable to handle kernel paging request at virtual address ffffffec
Mar  7 11:56:46 octopi kernel: [ 1483.810540] pgd = 80004000
Mar  7 11:56:46 octopi kernel: [ 1483.810549] [ffffffec] *pgd=36ffa821, *pte=00000000, *ppte=00000000
Mar  7 11:56:46 octopi kernel: [ 1483.810555] Internal error: Oops: 17 [#2] SMP ARM
Mar  7 11:56:46 octopi kernel: [ 1483.810586] Modules linked in: bnep hci_uart btbcm bluetooth brcmfmac brcmutil cfg80211 ftdi_sio rfkill joydev evdev usbserial snd_bcm2835 snd_pcm bcm2835_gpiomem snd_timer bcm2835_rng snd uio_pdrv_genirq uio ipv6
Mar  7 11:56:46 octopi kernel: [ 1483.810593] CPU: 1 PID: 42 Comm: kworker/1:1 Tainted: G      D         4.1.18-v7+ #846
Mar  7 11:56:46 octopi kernel: [ 1483.810594] Hardware name: BCM2709
Mar  7 11:56:46 octopi kernel: [ 1483.810602] task: b5b68000 ti: b5b2a000 task.ti: b5b2a000
Mar  7 11:56:46 octopi kernel: [ 1483.810613] PC is at kthread_data+0x18/0x20
Mar  7 11:56:46 octopi kernel: [ 1483.810619] LR is at wq_worker_sleeping+0x1c/0xe0
Mar  7 11:56:46 octopi kernel: [ 1483.810623] pc : [<80043d0c>]    lr : [<8003ecf4>]    psr: 00000193
Mar  7 11:56:46 octopi kernel: [ 1483.810623] sp : b5b2ba08  ip : b5b2ba18  fp : b5b2ba14
Mar  7 11:56:46 octopi kernel: [ 1483.810626] r10: 367c8000  r9 : b5b682ec  r8 : 8081a0a4
Mar  7 11:56:46 octopi kernel: [ 1483.810628] r7 : 807f8580  r6 : 807f8580  r5 : b6fc0580  r4 : 00000001
Mar  7 11:56:46 octopi kernel: [ 1483.810631] r3 : 00000000  r2 : 00000000  r1 : 00000001  r0 : b5b68000
Mar  7 11:56:46 octopi kernel: [ 1483.810635] Flags: nzcv  IRQs off  FIQs on  Mode SVC_32  ISA ARM  Segment user
Mar  7 11:56:46 octopi kernel: [ 1483.810638] Control: 10c5383d  Table: 32da806a  DAC: 00000015
Mar  7 11:56:46 octopi kernel: [ 1483.810640] Process kworker/1:1 (pid: 42, stack limit = 0xb5b2a210)
Mar  7 11:56:46 octopi kernel: [ 1483.810643] Stack: (0xb5b2ba08 to 0xb5b2c000)
Mar  7 11:56:46 octopi kernel: [ 1483.810648] ba00:                   b5b2ba2c b5b2ba18 8003ecf4 80043d00 b5b68000 b6fc0580
Mar  7 11:56:46 octopi kernel: [ 1483.810654] ba20: b5b2ba7c b5b2ba30 80569f88 8003ece4 800775cc 8007729c b5b2ba94 b5b2ba48
Mar  7 11:56:46 octopi kernel: [ 1483.810659] ba40: 80027e54 800775b4 00000001 807f4240 b5b38124 b5b2a000 b5b2b81c b5b2baa0
Mar  7 11:56:46 octopi kernel: [ 1483.810665] ba60: b5b68268 00000001 00000001 b5900000 b5b2ba94 b5b2ba80 8056a264 80569bb0
Mar  7 11:56:46 octopi kernel: [ 1483.810671] ba80: 0420806c b5b68000 b5b2bad4 b5b2ba98 800299f8 8056a224 b5b2bac4 80824dcc
Mar  7 11:56:46 octopi kernel: [ 1483.810677] baa0: b5b2baa0 b5b2baa0 8081dad4 808822c4 60000113 8081dad4 0000000b 7f12eb4c
Mar  7 11:56:46 octopi kernel: [ 1483.810682] bac0: 00000001 7f12eb4a b5b2bb5c b5b2bad8 80014138 800292a4 b5b2a210 0000000b
Mar  7 11:56:46 octopi kernel: [ 1483.810687] bae0: 00000000 8067a6f8 00000008 00000000 00000000 8081dad4 3006d06c 66666661
Mar  7 11:56:46 octopi kernel: [ 1483.810693] bb00: 20316666 32353365 35313030 66613920 65666666 35652066 30303164 28203230
Mar  7 11:56:46 octopi kernel: [ 1483.810699] bb20: 33393565 30303031 80002029 80568168 806be44c 00000000 00000017 b5b2bc80
Mar  7 11:56:46 octopi kernel: [ 1483.810704] bb40: 00000000 b5b68000 00000000 00000064 b5b2bb74 b5b2bb60 80567ea4 80013ef0
Mar  7 11:56:46 octopi kernel: [ 1483.810709] bb60: b5b2bc80 00000000 b5b2bbcc b5b2bb78 8056ed94 80567e3c ffffffff 80054460
Mar  7 11:56:46 octopi kernel: [ 1483.810715] bb80: b5b68048 80054490 b5b68048 b6fc05c0 00004017 00000000 b5b2bbdc b5b2bba8
Mar  7 11:56:46 octopi kernel: [ 1483.810720] bba0: 000003ff 8081e084 00000017 8056eb08 00000000 b5b2bc80 00000000 00000064
Mar  7 11:56:46 octopi kernel: [ 1483.810726] bbc0: b5b2bc7c b5b2bbd0 800091e8 8056eb14 b5b2bc5c b5b2bbe0 80057cc0 80056a28
Mar  7 11:56:46 octopi kernel: [ 1483.810731] bbe0: b6fc0580 00000002 b5b2bc2c 80055218 8008e8b0 80450f0c ffffffff b5b68048
Mar  7 11:56:46 octopi kernel: [ 1483.810737] bc00: b2e8e7c8 b5b68000 00000000 8005522c b5b68048 b2e8e7c8 b5b2bc54 b5b2bc28
Mar  7 11:56:46 octopi kernel: [ 1483.810742] bc20: 8005522c 80053f08 00000010 00000000 b5b68000 b6fc0580 b2e8e780 00000000
Mar  7 11:56:46 octopi kernel: [ 1483.810748] bc40: 00000001 8067ad70 b5b2bc6c b5b2bc58 8004cf9c 7f12eb48 60000013 7f12eb48
Mar  7 11:56:46 octopi kernel: [ 1483.810753] bc60: 60000013 ffffffff b5b2bcb4 00000000 b5b2bcec b5b2bc80 8056e2dc 800091ac
Mar  7 11:56:46 octopi kernel: [ 1483.810758] bc80: 0000000d b3f90084 000000c5 00000000 b3f90095 b2d60160 000000b4 000000c5
Mar  7 11:56:46 octopi kernel: [ 1483.810764] bca0: 00000000 00000000 00000064 b5b2bcec 00000003 b5b2bcc8 fffffff6 7f12eb48
Mar  7 11:56:46 octopi kernel: [ 1483.810769] bcc0: 60000013 ffffffff 00000000 b2d60160 000000c5 000000c5 00000000 00000000
Mar  7 11:56:46 octopi kernel: [ 1483.810774] bce0: b5b2bd74 b5b2bcf0 7f12fa18 7f12ea5c 80819cdc 00000c11 00000000 00000000
Mar  7 11:56:46 octopi kernel: [ 1483.810779] bd00: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
Mar  7 11:56:46 octopi kernel: [ 1483.810784] bd20: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
Mar  7 11:56:46 octopi kernel: [ 1483.810790] bd40: 00000000 00000000 b5247c40 b2d60160 000000d0 000000c5 00000064 00000c11
Mar  7 11:56:46 octopi kernel: [ 1483.810796] bd60: 00000000 b3f90004 b5b2bde4 b5b2bd78 7f251700 7f12f9b4 b3f9000c b2d60160
Mar  7 11:56:46 octopi kernel: [ 1483.810802] bd80: 00000000 00000000 00000c11 00000064 b3f90084 000000c5 ffffeae8 000000d0
Mar  7 11:56:46 octopi kernel: [ 1483.810807] bda0: 8056be54 00000c11 00000800 7f2598a4 b5b2bde4 b5b2bdc0 7f2598a4 b3f90004
Mar  7 11:56:46 octopi kernel: [ 1483.810812] bdc0: b3f90000 b5247c40 00000000 b2d60160 b2e3b248 b2e3b000 b5b2be0c b5b2bde8
Mar  7 11:56:46 octopi kernel: [ 1483.810818] bde0: 7f252204 7f25163c b5b2be88 b2d60340 b5b2bea0 b2d60340 b5247800 b5247c40
Mar  7 11:56:46 octopi kernel: [ 1483.810823] be00: b5b2be5c b5b2be10 7f252474 7f25219c 80056f60 80056a28 ffffffff 00ffffff
Mar  7 11:56:46 octopi kernel: [ 1483.810829] be20: 00000001 b6fc0ac0 0001ce82 00000001 b6fc0ad0 00000013 b2c36098 b2c360a8
Mar  7 11:56:46 octopi kernel: [ 1483.810834] be40: 00200200 00100100 b2c360ac b2c34000 b5b2be74 b5b2be60 7f259de0 7f2523c8
Mar  7 11:56:46 octopi kernel: [ 1483.810840] be60: b2de3cc3 b2de3c80 b5b2bee4 b5b2be78 7f259ea4 7f259da4 b529a8c0 b2e8e780
Mar  7 11:56:46 octopi kernel: [ 1483.810845] be80: b6fc0580 00000002 00000002 00000013 00000000 00000001 00000000 00000000
Mar  7 11:56:46 octopi kernel: [ 1483.810851] bea0: e5ef7b10 6c77669d 00000030 00000000 00000000 00000000 80569d94 b2c36098
Mar  7 11:56:46 octopi kernel: [ 1483.810856] bec0: b5b15a00 b6fc0094 b6fc0080 b6fc3f00 00000000 00000000 b5b2bf24 b5b2bee8
Mar  7 11:56:46 octopi kernel: [ 1483.810862] bee0: 8003dbf8 7f259e48 b6fc0080 b6fc0094 b5b2a000 00000008 b5b15a00 b6fc0080
Mar  7 11:56:46 octopi kernel: [ 1483.810867] bf00: b5b15a18 b6fc0094 b5b2a000 00000008 b5b15a00 b6fc0080 b5b2bf5c b5b2bf28
Mar  7 11:56:46 octopi kernel: [ 1483.810872] bf20: 8003dec4 8003dab0 8003de70 80818900 00000000 00000000 b5978080 b5b15a00
Mar  7 11:56:46 octopi kernel: [ 1483.810878] bf40: 8003de70 00000000 00000000 00000000 b5b2bfac b5b2bf60 8004364c 8003de7c
Mar  7 11:56:46 octopi kernel: [ 1483.810883] bf60: b5b2bf94 00000000 8004a260 b5b15a00 00000000 00000000 b5b2bf78 b5b2bf78
Mar  7 11:56:46 octopi kernel: [ 1483.810889] bf80: 00000001 00010001 b5b2bf88 b5b2bf88 b5978080 80043560 00000000 00000000
Mar  7 11:56:46 octopi kernel: [ 1483.810894] bfa0: 00000000 b5b2bfb0 8000faf8 8004356c 00000000 00000000 00000000 00000000
Mar  7 11:56:46 octopi kernel: [ 1483.810899] bfc0: 00000000 00000000 00000000 00000000 00000000 00000000 00000000 00000000
Mar  7 11:56:46 octopi kernel: [ 1483.810904] bfe0: 00000000 00000000 00000000 00000000 00000013 00000000 55555575 55555755
Mar  7 11:56:46 octopi kernel: [ 1483.810918] [<80043d0c>] (kthread_data) from [<8003ecf4>] (wq_worker_sleeping+0x1c/0xe0)
Mar  7 11:56:46 octopi kernel: [ 1483.810931] [<8003ecf4>] (wq_worker_sleeping) from [<80569f88>] (__schedule+0x3e4/0x674)
Mar  7 11:56:46 octopi kernel: [ 1483.810939] [<80569f88>] (__schedule) from [<8056a264>] (schedule+0x4c/0xa4)
Mar  7 11:56:46 octopi kernel: [ 1483.810947] [<8056a264>] (schedule) from [<800299f8>] (do_exit+0x760/0x9f8)
Mar  7 11:56:46 octopi kernel: [ 1483.810957] [<800299f8>] (do_exit) from [<80014138>] (die+0x254/0x43c)
Mar  7 11:56:46 octopi kernel: [ 1483.810965] [<80014138>] (die) from [<80567ea4>] (__do_kernel_fault.part.0+0x74/0x84)
Mar  7 11:56:46 octopi kernel: [ 1483.810973] [<80567ea4>] (__do_kernel_fault.part.0) from [<8056ed94>] (do_page_fault+0x28c/0x398)
Mar  7 11:56:46 octopi kernel: [ 1483.810980] [<8056ed94>] (do_page_fault) from [<800091e8>] (do_DataAbort+0x48/0xc4)
Mar  7 11:56:46 octopi kernel: [ 1483.810986] [<800091e8>] (do_DataAbort) from [<8056e2dc>] (__dabt_svc+0x3c/0x60)
Mar  7 11:56:46 octopi kernel: [ 1483.810988] Exception stack(0xb5b2bc80 to 0xb5b2bcc8)
Mar  7 11:56:46 octopi kernel: [ 1483.810994] bc80: 0000000d b3f90084 000000c5 00000000 b3f90095 b2d60160 000000b4 000000c5
Mar  7 11:56:46 octopi kernel: [ 1483.810999] bca0: 00000000 00000000 00000064 b5b2bcec 00000003 b5b2bcc8 fffffff6 7f12eb48
Mar  7 11:56:46 octopi kernel: [ 1483.811002] bcc0: 60000013 ffffffff
Mar  7 11:56:46 octopi kernel: [ 1483.811032] [<8056e2dc>] (__dabt_svc) from [<7f12eb48>] (cfg80211_get_bss_channel+0xf8/0x190 [cfg80211])
Mar  7 11:56:46 octopi kernel: [ 1483.811075] [<7f12eb48>] (cfg80211_get_bss_channel [cfg80211]) from [<7f12fa18>] (cfg80211_inform_bss_width+0x70/0x234 [cfg80211])
Mar  7 11:56:46 octopi kernel: [ 1483.811111] [<7f12fa18>] (cfg80211_inform_bss_width [cfg80211]) from [<7f251700>] (brcmf_inform_single_bss+0xd0/0x140 [brcmfmac])
Mar  7 11:56:46 octopi kernel: [ 1483.811140] [<7f251700>] (brcmf_inform_single_bss [brcmfmac]) from [<7f252204>] (brcmf_update_bss_info.part.8+0x74/0x130 [brcmfmac])
Mar  7 11:56:46 octopi kernel: [ 1483.811169] [<7f252204>] (brcmf_update_bss_info.part.8 [brcmfmac]) from [<7f252474>] (brcmf_notify_roaming_status+0xb8/0x1a0 [brcmfmac])
Mar  7 11:56:46 octopi kernel: [ 1483.811202] [<7f252474>] (brcmf_notify_roaming_status [brcmfmac]) from [<7f259de0>] (brcmf_fweh_call_event_handler+0x48/0xa4 [brcmfmac])
Mar  7 11:56:46 octopi kernel: [ 1483.811240] [<7f259de0>] (brcmf_fweh_call_event_handler [brcmfmac]) from [<7f259ea4>] (brcmf_fweh_event_worker+0x68/0x2d0 [brcmfmac])
Mar  7 11:56:46 octopi kernel: [ 1483.811264] [<7f259ea4>] (brcmf_fweh_event_worker [brcmfmac]) from [<8003dbf8>] (process_one_work+0x154/0x3cc)
Mar  7 11:56:46 octopi kernel: [ 1483.811272] [<8003dbf8>] (process_one_work) from [<8003dec4>] (worker_thread+0x54/0x500)
Mar  7 11:56:46 octopi kernel: [ 1483.811280] [<8003dec4>] (worker_thread) from [<8004364c>] (kthread+0xec/0x104)
Mar  7 11:56:46 octopi kernel: [ 1483.811288] [<8004364c>] (kthread) from [<8000faf8>] (ret_from_fork+0x14/0x3c)
Mar  7 11:56:46 octopi kernel: [ 1483.811294] Code: e24cb004 e52de004 e8bd4000 e59032c0 (e5130014) 
Mar  7 11:56:46 octopi kernel: [ 1483.811297] ---[ end trace b08645ac099d9c90 ]---
Mar  7 11:56:46 octopi kernel: [ 1483.811299] Fixing recursive fault but reboot is needed!
```

Big kernel oops there. Apparently in the wifi driver. Did I mention that I have a feeling that the drivers for the included wifi chipset aren't quite there yet?

**Comment by A-Nesh** [comment]
@foosel do you think that using a USB WIFI adapter will solv the issue?

**Comment by foosel** [comment]
Absolutely no idea. Might be worth a try. Maybe even blacklisting the driver of the built-in wifi.

What will definitely solve a lot though is the RPi people fixing their drivers ;)

**Comment by A-Nesh** [comment]
I can confirm now that blacklisting the `brcmfmac` driver solvs the issue for me. It now works flawlessly with a USB WiFi adapter.

**Comment by foosel** [comment]
@A-Nesh I forwarded this thread to someone at raspberry pi, the engineering team suggested a different power supply, replied that you already tried that, they also suggested to try stock raspbian - just for the record, when you talked about the "full image" further up, did you mean stock raspbian/noobs?

**Comment by A-Nesh** [comment]
@foosel I have used this:
RASPBIAN JESSIE
Full desktop image based on Debian Jessie
Version:February 2016
Release date:2016-02-26

I have tested 4 different power supplies, 2 SD cards, tested setting `max_usb_current=1`and `wireless-power off`... it freeze after a while with and without 3D printer connected. 

Since i have blacklisted `brcmfmac` it printed flawlessly with 1.0A chromecast power supply while running `hardinfo` :)

**Comment by techied** [comment]
;D so happy I discovered this thread
Can confirm that latest octopi (not nightly) does not work on my brand-new pi 3 and stock psu
Will follow directions here and either use nightly build or Jessie =D

**Comment by A-Nesh** [comment]
i have now switched to octopi nightly build and blacklisted `brcmfmac`. It has now been printing for two hours without problems.

**Comment by techied** [comment]
It works with the latest nightly build (released on the ninth of March) 
Haven't tried printing yet, but thanks for the help =D

**Comment by techied** [comment]
Finished a print on the nightly build no problem :D 
I used the stock wifi and camera module, and a 2.4amp psu
no crashes or anything :)

**Comment by A-Nesh** [comment]
Yeah, it just proves that i have some kind of hardware issue with my WiFi chip.

**Comment by picclock** [comment]
Brilliant helpful thread. Same issues with RPi3, been driving me mad. Have a wifi dongle but how do you blacklist the brcmfmac file, or is there an image with it blacklisted?

**Comment by foosel** [comment]
@picclock have you the same crashing issues as @A-Nesh? You definitely should verify first that it isn't just your Wi-Fi going to sleep by disabling it's power management as described [above](https://github.com/guysoft/OctoPi/issues/207#issuecomment-192938770). Only if that doesn't work should you test if black listing the module works. To black list it,

**Comment by foosel** [comment]
Argh, misclicked, sorry

**Comment by picclock** [comment]
Hi Foosel, Thx for the help. I will do the wifi power bit. I can use a LAN connection to edit it as the wifi is all over the place. Am running 2016-02-26-octopi-jessie-lite-0.13.0.img. 
Many Thx

**Comment by A-Nesh** [comment]
Yes, i have tried to disable wifi power management with no sucess. It completely freezes, does not recognize monitor and keyboard. The only solution i found working for me is blacklisting `brcmfmac `in `/etc/modprobe.d/raspi-blacklist.conf` 

On the bright side OctoPi nigtly build works great, has been printing with it for ower a week now.

**Comment by picclock** [comment]
@ A.Nesh
Exactly the same experience. Followed instructions to disable wifi power mgmt and no wifi. Plugged in LAN cable, ssh, restored backup interfaces file, blacklisted driver and plugged in wifi adapter. Rock solid. !!!!  

Clearly some issues that need sorting.

Many , many thx

best regards

picclock

**Comment by foosel** [comment]
@picclock I'm still not 100% sure what problem exactly you are experiencing.

Did you have kernel panics like @A-Nesh (full out system crashes) or not?

There are two issues here:

  1. One manifests as the Pi still being alive but the wifi going down after some idle time due to power management kicking in (and being broken). That can be fixed by disabling power management by adding `wireless-power off` to the wifi config in `/boot/octopi-network.txt`. Apparently there is also a fix available through `rpi-update` or possibly even `apt-get upgrade` by now, but personally I'd prefer a fixed raspbian base image...

  2. The other is a severe bug in the driver causing a full out kernel panic in some cases, effectively locking down the system (and causing such nice things as prints stopping and heaters staying on). That manifests as the Pi not even reacting to a wired connection, keyboard, monitor etc but should hopefully be visible in `/var/log/syslog`. This can be solved by blacklisting the `brcmfmac` kernel module, but that also removes the possibility to use the integrated wifi at all (you need to use a wired connection then, or an external wifi dongle). It's a bit like solving an issue with a home appliance not by repairing it but throwing it on the trash and buying a new one - it's not a fix.

I'm asking to be very specific as to what issue specifically you are seeing (one or two or both), because the RPi people I've been talking to over some hops in between appear not be convinced yet that there are more issues than the power management related one (number 1), so any proofs for the existence of issue number 2 (in the form of hardware setup used + syslog with kernel panic etc) would help to hopefully raise awareness and get this issue addressed as well.

And just for the record, issue number 2 is nothing that can be addressed through OctoPi, it's something that needs to be fixed in the kernel driver for the wifi chipset, so something that needs to be done by the people maintaining that driver.

**Comment by picclock** [comment]
Hi Foosel. Thx for your efforts. I disabled the wifi power save as in 
http://www.modmypi.com/blog/disable-wifi-power-management
After this the pi would not connect to the router. It could not be seen at all. I then replaced the file I had edited with the backup, and blacklisted the brcmfmac file and it worked and has been running without issues since. 

It may be that the wifi power mod is incorrect, or, because I deleted the file contents and replaced them with a paste from the web page, that I may have deleted something else which was needed.

The initial problems were that wifi connected intermitantly, then would no longer connect at all. 

If you wish me to run any tests to diagnose I wil be happy to do so, but please make it simple !!

Best Regards

picclock

**Comment by picclock** [comment]
@ Foosel
OK, I just disabled the wifi power ( in the octopi-network, added  wireless-power off) and it seems good so far without the dongle. Will keep you updated if it fails.

Best Regards

picclock

**Comment by foosel** [comment]
@picclock then blacklisting the module is possibly not necessary

----

#### General instructions on how to test for wifi issues with the RPi3 here

Download and flash a fresh OctoPi *0.13 nightly* image (0.12 doesn't work and will give you the "rainbow image"). Configure your wifi credentials while still using the SD card as a thumb drive while still in the computer you flashed it from by editing `octopi-network.txt` in the root of that thumb drive. Make sure your text editor doesn't mangle the quotes around the ssid and passphrase (especially TextMate on Mac appears to love to do that, you want " or ' if either ssid or passphrase contains spaces, nothing else!). Plug the SD into the Pi, do NOT attach a wifi dongle. Boot the Pi, verify that it connects to your network. Then observe its long time behaviour while using it (e.g. printing stuff).

If an issue arises, is it

a) the Pi stopping to react to network access via wifi but still apparently running (e.g. if a print is running it continues, a keyboard and monitor still work, ...) or

b) the Pi stopping to react to network access and anything else, apparently crashing (e.g. if a print is running it stops, a keyboard and monitor don't still work, ...) or

c) nothing of the above

**If a)**, does this solve the issue?

  1. Shutdown the Pi, unplug the SD, plug it into your computer's card reader
  2. Edit `octopi-network.txt` again
  3. Turn

     ```
     ## WPA/WPA2 secured
     iface wlan0-octopi inet manual
         wpa-ssid "YOUR SSID"
         wpa-psk "YOUR PASSWORD"
     ```

     into 

     ```
     ## WPA/WPA2 secured
     iface wlan0-octopi inet manual
         wpa-ssid "YOUR SSID"
         wpa-psk "YOUR PASSWORD"
         wireless-power off
     ```
     and save.
  4. Put the SD back into the Pi, reboot, observe

**If b)**, please provide your `/var/log/syslog` file.

**If c)**, sounds like a so far unknown issue.

If you are only experiencing a), the solution to a) should work. If you are not seeing your Pi3 connect  to your network after applying the solution to a), something else is broken in your configuration that has nothing to do with either a) or b) (probably a mistyped password, mangled quotes or something similar).

Please do not try to apply other steps outlined in other blog posts around the internet, follow the steps outlined in this thread. Give precise feedback as to what exact behaviour you are observing at all times.

**Comment by ghollingworth** [comment]
Hi,

Gordon Hollingworth here from Raspberry Pi.  The WiFi power management issue has now been fixed, apt-get update / upgrade on a Raspbian distribution will fix this for you.  I'd suggest pulling that kernel into the OctoPi distribution.

The driver crashing issue is specific to this case only.  We're not seeing this anywhere else so it is somewhat strange (not saying it isn't a problem just that it's going to be hard to understand / diagnose!)

Could do with answers to the following:

1) If you return to Raspbian Jessie and sudo apt-get upgrade / update
2) edit /boot/config.txt and change arm_freq to 600 (i.e. force underclock just in case)
3) Does this still cause the problem
4) Other possibility is that it's something to do with the WiFi access point settings...  Can you try forcing the channel to somewhere in the middle of the range and turn off encryption for a little while to see if's to do with that?

Gordon

**Comment by A-Nesh** [comment]
@ghollingworth i'll try it later today and get back to you

**Comment by foosel** [comment]
@ghollingworth thanks for reaching out. We can add the kernel update step to the OctoPi build script, but would prefer to base OctoPi on a Raspbian release that already includes that. Do you know if a new stable image release is planned soon? If not we'll have to go with adding the upgrade (in the hopes that this works with our qemu based build).

**Comment by ghollingworth** [comment]
We are planning on a new update soon, but currently no date, should be in
the next week or so�

Gordon

-- 
Director of Engineering, Raspberry Pi (Trading) Limited
T:	+44 (0) 1223 322633
M:	+44 (0) 7450 946289
E:	gordon@raspberrypi.org
W:	www.raspberrypi.org

**Comment by foosel** [comment]
Thank you, that's something to go on.

**Comment by seabas917** [comment]
Hey guys, 
Am I crazy or is the latest published nightly (2016-03-15_2016-02-26-octopi-jessie-lite-0.13.0) missing its Octoprint install?

On the other hand it looks like after ssh-ing in wifi credentials, the onboard wifi is working really well! Same ping quality as my laptop, in this build. I'll leave ping test up all night to see if this is stable. I'll do a manual Octoprint install after seeing if the system stays up.

Thanks Foosel and Gordon! 
Amazing work from both of you!

Side note: No surprise here, but the AdafruitLCD install script in Nightly(03-15) for the OctoPanel, will crash the Pi3 system on next reboot and not recover.. 

The boot/octopi-network file isn't working, so i suggest manual setup for onboard wifi:
`sudo nano /etc/wpa_supplicant/wpa_supplicant.conf `

and adding a WPA2 network:
```
network={
    ssid=“*****”
    psk="*****"
}
```
Reboot Wifi:
```
sudo ifdown wlan0
sudo ifup wlan0
```
Then confirming Wifi IP with : `ifconfig`

**Comment by guysoft** [comment]
Yes! yesterday nightly is bad, don't use it

**Comment by guysoft** [comment]
@seabas917 There is a new nightly build, should work now

**Comment by picclock** [comment]
Hi I have just tried the new build and it has broken some menu items in the
system menu with an API error.

changes as listed below :-

sudo nano /etc/init.d/octoprint
following at line 77, before RETVAL="$?"

gpio export 17 out
gpio export 27 out
gpio export 22 out

And :-

sudo nano ~/.octoprint/config.yaml

and in the system>actions section, after actions:-

  - action: printer on
    command: gpio -g write 27 1
    name: Printer on
  - action: printer off
    command: gpio -g write 27 0
    name: Printer off
  - action: camera on
    command: gpio -g write 17 1
    name: Camera on
  - action: camera off
    command: gpio -g write 17 0
    name: Camera off
  - action: aux on
    command: gpio -g write 22 1
    name: Aux on
  - action: aux off
    command: gpio -g write 22 0
    name: Aux off

Hopefully can be easily fixed:

Best Regards

oldun

**Comment by A-Nesh** [comment]
@ghollingworth i did what you said:
did `sudo apt-get upgrade` / `update`
changed `arm_freq` to `600` was `800` before!?
and unblocked `brcmfmac` 

I left it over night, when i got home it was dead. both LEDs (red end green) are on and it not responds on keyboard or monitor.

The `syslog` did not reveal anything interesting other than some error. 

`Mar` 16 06:17:05 octopi systemd[1]: Starting Copy rules generated while the root was ro...
Mar 16 06:17:05 octopi systemd[1]: Starting LSB: Set preliminary keymap...
Mar 16 06:17:05 octopi systemd[1]: Started Copy rules generated while the root was ro.
Mar 16 06:17:05 octopi systemd-fsck[71]: e2fsck 1.42.12 (29-Aug-2014)
Mar 16 06:17:05 octopi systemd-fsck[71]: /dev/mmcblk0p2: clean, 50245/1881152 files, 477961/7775360 blocks
Mar 16 06:17:05 octopi fake-hwclock[77]: Wed 16 Mar 06:17:01 UTC 2016
Mar 16 06:17:05 octopi systemd[1]: Found device /dev/ttyS0.
Mar 16 06:17:05 octopi systemd[1]: Starting Sound Card.
Mar 16 06:17:05 octopi systemd[1]: Reached target Sound Card.
Mar 16 06:17:05 octopi systemd-udevd[120]: Error calling EVIOCSKEYCODE on device node '/dev/input/event0' (scan code 0xc00b8, key code 161): Invalid argument
Mar 16 06:17:05 octopi systemd-udevd[120]: Error calling EVIOCSKEYCODE on device node '/dev/input/event0' (scan code 0xc0183, key code 226): Invalid argument
Mar 16 06:17:05 octopi systemd-udevd[120]: Error calling EVIOCSKEYCODE on device node '/dev/input/event0' (scan code 0xc0184, key code 421): Invalid argument
Mar 16 06:17:05 octopi systemd-udevd[120]: Error calling EVIOCSKEYCODE on device node '/dev/input/event0' (scan code 0xc0186, key code 423): Invalid argument
Mar 16 06:17:05 octopi systemd-udevd[120]: Error calling EVIOCSKEYCODE on device node '/dev/input/event0' (scan code 0xc018a, key code 155): Invalid argument
Mar 16 06:17:05 octopi systemd-udevd[120]: Error calling EVIOCSKEYCODE on device node '/dev/input/event0' (scan code 0xc018e, key code 397): Invalid argument
Mar 16 06:17:05 octopi systemd-udevd[120]: Error calling EVIOCSKEYCODE on device node '/dev/input/event0' (scan code 0xc01b6, key code 212): Invalid argument
Mar 16 06:17:05 octopi systemd-udevd[120]: Error calling EVIOCSKEYCODE on device node '/dev/input/event0' (scan code 0xc01bc, key code 430): Invalid argument
Mar 16 06:17:05 octopi systemd-udevd[120]: Error calling EVIOCSKEYCODE on device node '/dev/input/event0' (scan code 0xc0221, key code 217): Invalid argument
Mar 16 06:17:05 octopi systemd-udevd[120]: Error calling EVIOCSKEYCODE on device node '/dev/input/event0' (scan code 0xc0223, key code 172): Invalid argument
Mar 16 06:17:05 octopi systemd-udevd[120]: Error calling EVIOCSKEYCODE on device node '/dev/input/event0' (scan code 0xc022d, key code 418): Invalid argument
Mar 16 06:17:05 octopi systemd-udevd[120]: Error calling EVIOCSKEYCODE on device node '/dev/input/event0' (scan code 0xc022e, key code 419): Invalid argument
Mar 16 06:17:05 octopi systemd[1]: Found device /dev/mmcblk0p1.
Mar 16 06:17:05 octopi systemd[1]: Starting File System Check on /dev/mmcblk0p1...
Mar 16 06:17:05 octopi keyboard-setup[108]: Setting preliminary keymap...done.
`
No traces of crush or freeze...
the end of syslog looks like this:

`Mar 16 06:17:13 octopi systemd[1]: Starting HAProxy Load Balancer...
Mar 16 06:17:13 octopi systemd[1]: Starting /etc/rc.local Compatibility...
Mar 16 06:17:13 octopi systemd[1]: Starting Network is Online.
Mar 16 06:17:13 octopi systemd[1]: Reached target Network is Online.
Mar 16 06:17:13 octopi systemd[1]: Starting LSB: Start NTP daemon...
Mar 16 06:17:13 octopi systemd[1]: Started /etc/rc.local Compatibility.
Mar 16 06:17:13 octopi systemd[1]: Starting Wait for Plymouth Boot Screen to Quit...
Mar 16 06:17:13 octopi systemd[1]: Starting Terminate Plymouth Boot Screen...
Mar 16 06:17:13 octopi systemd[1]: Started Wait for Plymouth Boot Screen to Quit.
Mar 16 06:17:13 octopi systemd[1]: Started Terminate Plymouth Boot Screen.
Mar 16 06:17:13 octopi systemd[1]: Starting Getty on tty1...
Mar 16 06:17:13 octopi systemd[1]: Started Getty on tty1.
Mar 16 06:17:13 octopi systemd[1]: Starting Serial Getty on ttyS0...
Mar 16 06:17:13 octopi systemd[1]: Started Serial Getty on ttyS0.
Mar 16 06:17:13 octopi systemd[1]: Starting Login Prompts.
Mar 16 06:17:13 octopi systemd[1]: Reached target Login Prompts.
Mar 16 06:17:13 octopi ntpd[725]: ntpd 4.2.6p5@1.2349-o Mon Nov  2 04:29:47 UTC 2015 (1)
Mar 16 06:17:13 octopi ntpd[728]: proto: precision = 0.573 usec
Mar 16 06:17:13 octopi ntpd[728]: Listen and drop on 0 v4wildcard 0.0.0.0 UDP 123
Mar 16 06:17:13 octopi ntp[712]: Starting NTP server: ntpd.
Mar 16 06:17:13 octopi systemd[1]: Started LSB: Start NTP daemon.
Mar 16 06:17:13 octopi ntpd[728]: Listen and drop on 1 v6wildcard :: UDP 123
Mar 16 06:17:13 octopi ntpd[728]: Listen normally on 2 lo 127.0.0.1 UDP 123
Mar 16 06:17:13 octopi ntpd[728]: Listen normally on 3 wlan0 192.168.1.88 UDP 123
Mar 16 06:17:13 octopi ntpd[728]: Listen normally on 4 wlan0 192.168.1.232 UDP 123
Mar 16 06:17:13 octopi ntpd[728]: Listen normally on 5 lo ::1 UDP 123
Mar 16 06:17:13 octopi ntpd[728]: Listen normally on 6 wlan0 fe80::ba27:ebff:fe96:95d5 UDP 123
Mar 16 06:17:13 octopi ntpd[728]: peers refreshed
Mar 16 06:17:13 octopi ntpd[728]: Listening on routing socket on fd #23 for interface updates
Mar 16 06:17:13 octopi systemd[1]: Started HAProxy Load Balancer.
Mar 16 06:17:13 octopi systemd[1]: Starting Multi-User System.
Mar 16 06:17:13 octopi systemd[1]: Reached target Multi-User System.
Mar 16 06:17:13 octopi systemd[1]: Starting Update UTMP about System Runlevel Changes...
Mar 16 06:17:14 octopi systemd[1]: Started Update UTMP about System Runlevel Changes.
Mar 16 06:17:14 octopi systemd[1]: Startup finished in 3.525s (kernel) + 14.625s (userspace) = 18.150s.
Mar 16 06:17:18 octopi dhcpcd[708]: wlan0: no IPv6 Routers available
Mar 16 13:21:28 octopi systemd[1]: Time has been changed
Mar 16 13:36:04 octopi systemd[1]: Starting Cleanup of Temporary Directories...
Mar 16 13:36:04 octopi systemd[1]: Started Cleanup of Temporary Directories.`

i don't really understand why am i getting 2 IPs. a DHCP (232) and a static one (88)

**Comment by A-Nesh** [comment]
it crashed again, right now
![skarmavbild 2016-03-16 kl 14 56 37](https://cloud.githubusercontent.com/assets/17685372/13814655/6e29f388-eb87-11e5-978e-f95925912147.png)

**Comment by Cyphrz** [comment]
Just an update, a new base image is out

**Comment by guysoft** [comment]
There is already a new nightly build with the new image.

**Comment by yenoromp** [comment]
Hi
Ive just installed the latest daily but octopi does not start automatically in it.
Paul

**Comment by guysoft** [comment]
Bug in the nightly builds due to #214 , don't install it

**Comment by guysoft** [comment]
New nightly out with fix, @yenoromp you can try it now.

**Comment by yenoromp** [comment]
Thanks

I will give it a go tonight


Paul Moroney
0400720144

**Comment by yenoromp** [comment]
Hi
Tested it tonight and all ok.
it looks very good , will do some printing tonight.
Paul

**Comment by lee-thompson** [comment]
As someone who was reading this thread with interest (after not being able to even get my Octopi to be recognised on my network, and being totally new to all this), I can confirm that the most recent build worked for me on a Raspberry Pi 3. Thank you!

**Comment by WZ9V** [comment]
lee-thompson, what specific build did you use?

I'm looking to setup a RPi 3 with wireless OctoPi.

**Comment by lee-thompson** [comment]
@WZ9V I used the 2016-03-20_2016-03-18-octopi-jessie-lite-0.13.0-0.13.0.zip from the nightly builds (http://docstech.net/OctoPiMirror/nightly/) - and I had to do it twice because MacOS's default text editor replaced my quote marks with the _wrong_ kind when I updated my wifi credentials. Sublime Text was my saviour.

I literally punched the air when I could see it on my network though!

**Comment by WZ9V** [comment]
Thanks, gives me a good starting point and since I have Linux I don't expect any special quote issues.

**Comment by crazydart** [comment]
2016-03-23_2016-03-18-octopi-jessie-lite-0.13.0.zip (2016-03-23 00:36:17 UTC) is no good with the RPi3 and wifi.

**Comment by 3driffs** [comment]
2016-03-20_2016-03-18-octopi-jessie-lite-0.13.0-0.13.0.zip (2016-03-20 11:58:46 UTC)  prints, but I couldn't get the local GUI running.  After mucking about installing packages, I have a local GUI and browser, but the camera isn't showing up in octoprint.

Net:  I'm a LONG time removed from playing around at the UNIX/Linux level of an OS.  Any guess as to when we'll see a RPi3 compatible stable build?

**Comment by gddeen** [comment]
I believe jessie 'lite' was built to NOT have a desktop session...
An easier way would be to get the raspbian jessie build and then
follow the steps to build octopi.

If you DO have a desktop locally, I don't think Octoprint is designed
to run and display on an OctoPi. (I guess this means they didn't test it
or the proxy doesn't like it? I'm clueless.)

**Comment by foosel** [comment]
The Jessie Lite base image doesn't have X included, and considering how much bloat that whole desktop cruft adds without being needed for the actual usage scenario that's a very very good thing. 

> If you DO have a desktop locally, I don't think Octoprint is designed
to run and display on an OctoPi. (I guess this means they didn't test it
or the proxy doesn't like it? I'm clueless.)

"They" did and it works just fine but "they" do NOT recommend it (and in fact strongly discourage it) since running a full desktop environment with CPU and memory hogging components including a software rendering browser next to a critical real-time job such as feeding a file to a 3d printer is a recipe for tears. OctoPrint is designed to be accessed from a different computer than the one it's running on for a reason, which means a lot of resource hungry things happen on the client to keep load off the backend. If you run both on the same device, you throw away all advantages.

And now surely people will swoop in again and tell me "oh, but I run it just fine, the four cores are plenty and I NEVER saw any bad consequences by running my full blown desktop environment on my Pi and accessing OctoPrint through that while printing a 20 hour print job with lots of curves, you are wrong!" - and I can just say, YMMV, but I wouldn't risk it. 

Experience has shown that people will run way too much stuff next to OctoPrint on even a single core pi (ideally with an insufficient PSU for extra party) and then complain that it's Octo's fault their prints fail, so being able to build upon on a way less bloated base image that doesn't contain that particular risk in the first place has many advantages (download size being one of them btw, that stuff costs money...).

And if people desperately need X against all recommendations:

    $ sudo ~pi/scripts/install-desktop

**Comment by nicolinux** [comment]
You should add "server" next to OctoPrint's name so people won't get the wrong idea.
One does not run servers with X, except you really need it for the server's task, which for OctoPrint is not the case.

**Comment by foosel** [comment]
Wouldn't help ;)

**Comment by 3driffs** [comment]
Well, the stable build has both X and a browser as part of it.  I have a keyboard/monitor/mouse in the workroom with the printer, so I can operate it directly without fighting with the tiny LCD screen on the printer itself.

**Comment by foosel** [comment]
The stable build is based on Raspbian Wheezy, where no lite version without all the clutter they added (like Minecraft and whatnot, on top of X) was available. With 1.5GB base size.

The nightlies are based on Raspbian Jessie Lite.

**Comment by 3driffs** [comment]
So going forward, no built-in GUI capability.  Bummer - that's a big loss of functionality for me.  Guess I'll have to dust off my linux skills and build my own platform.

**Comment by foosel** [comment]
Or just install an X server? You know, via the existing package system? Or actually even something as simple as this (and yes, that is already included in the current 0.13 builds):

> And if people desperately need X against all recommendations:
>
>     $ sudo ~pi/scripts/install-desktop

That means you'll have to wait a bit while it downloads a bunch of packets that are not needed for the core use case, but that's a one time thing. 

And that you download these packets only if you need them instead of having them included in the image saves us time on every release trying to figure out what additional crud got added to the full Raspbian image that is not needed and only blows up the download size and saves everyone kind enough to provide a download mirror actual money for the saved traffic, not to mention saves everyone who just wants to run OctoPi as a headless server (the majority) the time needed to download the extra megabytes of image size they don't need for their use case. So win win for everyone really.

`apt-get` is there for a reason. Not every possible use case needs to be shipped out of the box if stuff is easy to install post-flash. Especially not if trying to do that doubles or triples the download size for everyone.

**Comment by 3driffs** [comment]
Thanks, I missed that in the earlier post.  No disagreement on the strategy, just wasn't aware it had changed.  I may yet just run two PI's (they're cheap enough), my old  2 as the dedicated server, and the 3 with a full desktop environment.  I haven't seen issues, but your point about load on during a long print is well taken.

**Comment by 3driffs** [comment]
And just to close out my part in the thread, two pi's works great.  Thanks for the info.

**Comment by yenoromp** [comment]
I'm sticking with the image I installed a week or so ago when the octopi install was fixed 
Done heaps of prints since then 


Paul Moroney
0400720144

**Comment by glm1157** [comment]
Which nightly are you using?  I tried the one of the latest builds (either 2016-04-04 or 2016-04-05, I forget) and I had problems setting up WiFi and connecting to my printer.  It might be me or the build.  Just want to try something that has worked for someone to eliminate one of the variables.

Thanks,

Gary

**Comment by yenoromp** [comment]
Mine is an oldish nightly 

I will check later , but if you check the size of the download , mine is the second one after the size increase .... And it's working fine 


Paul Moroney
0400720144

**Comment by glm1157** [comment]
Thanks.  I grabbed 2016-03-25_2016-03-18-octopi-jessie-lite-0.13.0.zip and had  the same problems.  I cannot get Wi-Fi to work.  I am using WEP which might be part of the problem.  Some folks have had problems with it on their Pi2s.

My second problem was fixed by switching to a lower baud rate.  I assumed that I could use 250000 like I did on my PogoPlug version of Octoprint.  I had to go down to 115200.  It's working now.  Printing a case for my PI now.

Thanks,

Gary

**Comment by yenoromp** [comment]
My wifi worked 

I used the old way without the supplicant , as well as edited the networks file directly ...


Paul Moroney
0400720144

**Comment by notcranium** [comment]
Hi.  Great job on what has been done so far.  I hope this keeps progressing.  I am supporting the project through Patreon!

I'm using jessie-lite-0.13.0.  I don't have my printer yet so can't post if there are issues with that for another week.  But the problems I see right now are:

- SD card is not initialized so I cannot upload files to it.  Selecting the option to initialize it doesn't do anything.
- Timelapse is completely greyed out.  The Logitech C310 appears to work like a champ though.
- I have a profile set up in the Cura Engine but can't seem to do anything with a .STL that I uploaded (maybe because there is no printer attached?) 

I'm completely new to this so maybe my issues are related to not really knowing what I'm doing.  :)

**Comment by nophead** [comment]
You can't upload to SD without a printer attached because the SD card it refers to is one attached to the printer. You also can't set the timelapse options without a printer attached. 

Not sure about slicing as I don't use it.

**Comment by SirVer** [comment]
I have a similar issue and found out that my raspberry does not like 2.4GHz channels 12 and 13 - if your router is set to AUTO choose the channels, this might be your issue. See https://github.com/raspberrypi/linux/issues/1442#issuecomment-214993828.

**Comment by yenoromp** [comment]
Hi

Could this to be with time zones as some channels are not allowed in certain countries ?


Paul Moroney
0400720144

**Comment by guysoft** [comment]
Fixed in 0.13.0

**Comment by alex-gibson** [comment]
Thanks to everyone who contributed to fixing this issue!

**Comment by SIXM6N** [comment]
After it just happened again, I have to agree, doesn't seem to be power related.

I also have to correct myself, it apparently doesn't freeze completely after all, it just looses wifi connectivity. Funny thing is, it believes it still does have connectivity, has an IP and everything, but cannot be pinged or SSH'd into. Until I ping the router from the Pi3, then suddenly connectivity is back:

$ ping -t 192.168.1.204

Pinging 192.168.1.204 with 32 bytes of data:
Reply from 192.168.1.3: Destination host unreachable.
Reply from 192.168.1.3: Destination host unreachable.
Reply from 192.168.1.3: Destination host unreachable.
Reply from 192.168.1.3: Destination host unreachable.
Reply from 192.168.1.204: bytes=32 time=2030ms TTL=64
Happens very reproducibly after some idle time. Maybe some power safe feature of the wifi chipset that's misbehaving (like it also happens with some dongles out there)?

@A-Nesh did it really freeze for you during printing (print stopped) or could it be that just the network connection broke down (print didn't stop, but anything relying on network didn't work anymore, so OctoPrint, SSH, ...).
I have pretty much tried everything and have condemned my Pi3 after the above happened to me as well Foosel. That was a copy paste of yours back in March by the way. But mine was identical. I was able to boot it move around in config but that was it. At this rate raspberry Pi is not on my good side. Octoprint was great at first and then the time lapse feature stopped working about a week ago and now the Pi has totally stopped communicating. Extremely frustrated. I reloaded the Pi several times and set it up for wifi. It fixed it the first time. But after that no more communication over the wifi. Below is my log for the time lapse issue at hand. Sorry to blow up this post. Maybe the the Pi was causing the time lapse failures. I'm a noob at this so please excuse my terminology if it fails you. I'm going to pick up another Pi this weekend unless you can convince me otherwise. I have tried the above network.text code that you added and I am running the latest and greatest (not beta). Thanks for your time.

https://db.tt/ehSBNU4a

**Comment by foosel** [comment]
@SIXM6N Your timelapse issues are caused by the path to ffmpeg being misconfigured. On OctoPi 0.13 (which is what you should be running) that should read `/usr/bin/avconv`. If you are still running into those WiFi power management issues with OctoPi 0.13 I have no idea. Stuff has been pretty solid for me since the kernel module got fixed, and apparently it's also been working great for the people out there. Maybe there's something wrong with your specific Pi after all.

**Comment by picclock** [comment]
Hi
To resolve my WiFi pi3 issues I used an external USB WiFi adapter which
sorted all the issues. I would guess this is selected in preference to the
built in one.
Hope this helps
Best Regards
picclock

**Comment by ghollingworth** [comment]
The Power management on the Pi3 WiFi is disabled in a downstream patch

https://github.com/raspberrypi/linux/commit/e1d6cd19b95f323a2f7b8bcf4c483a9326db1f98

Please check your dmesg for "power management disabled" to see if you have this.  If not then you need to update your linux kernel, modules and firmware

Gordon

**Comment by SIXM6N** [comment]
Okay thanks for looking into it for me. Very much appreciated.

**Comment by SIXM6N** [comment]
Okay I did get my Pi3 resurrected and communicating by installing a fresh download from the Octoprint site. So excited. 
 The line /usr/bin/avconv was changed because I read the change on a site that would put the files directly into the exact path on my PC. And it actually worked for about 5 prints. But after the clean install it actually works the way it is supposed to.
 I have another question about a post I read today about deleting a time lapse caused a deletion in the network.text file which leads me to believe that the Time lapses are stored on the SD. Is that true? Because I do remember deleting some time lapses just before my network stopped operating. My SD is only 30G so not much room if thats the case. I also noted that shutting down the power before shutting down the server through Octoprint 1st could corrupt it at well. At this point I'm not sure which one or maybe both caused the issue. I hope you can clarify this somewhat.

**Comment by foosel** [comment]
OctoPrint doesn't even have write access to the network configuration file ;) And no, timelapse management cannot cause that. Pulling the power though at the wrong moment can. Always shut down the Pi first via the System menu *and wait until it stops blinking* before pulling power.

In general, if you have any doubts/questions/etc, best get in touch on the [mailing list](https://groups.google.com/group/octoprint) or in the [G+ community](https://plus.google.com/communities/102771308349328485741), those are the official support channels. You'll get competent help there (as long as you provide enough information to get help of course ;)). There's a lot of misinformation across the net.
